File: api/src/main/java/org/mcphoton/runtime/Updatable.java
package org.mcphoton.runtime;

/**
 * @author TheElectronWill
 */
public interface Updatable {
	/**
	 * Performs an update.
	 *
	 * @param dt the elapsed time, in seconds, since the last execution of the update loop
	 */
	void update(double dt);
}

File: core/src/main/scala/org/tuubes/core/engine/Updatable.scala
package org.tuubes.core.engine

/**
 * Trait for updatables objects. Updates are based on the time elapsed since the last batch of
 * updates ("delta-time" based).
 *
 * @author TheElectronWill
 */
trait Updatable {
	def update(dt: Double): Unit
}